src/SFML/Window/WindowImpl.cpp

```diff
@@ -175,30 +176,50 @@ void WindowImpl::setMaximumSize(const std::optional<Vector2u>& maximumSize)
 
 
 ////////////////////////////////////////////////////////////
-Event WindowImpl::popEvent(bool block)
+Event WindowImpl::popEventBlocking(Time timeout)
 {
+    const auto startTime = std::chrono::steady_clock::now();
+
+    const auto timedOut = [&]
+    { return timeout != Time::Zero && (std::chrono::steady_clock::now() - startTime) < timeout.toDuration(); };
```

Comment:
I considered taking a `std::optional<Time>` instead of using the `Time::Zero` magic value, but it would have been inconsistent with any other timeout.

Reply:
Perhaps we revisit this library-wide in a later PR. Using `Time::Zero` as a sentinel value isn't something I love.

Reply:
> Perhaps we revisit this library-wide in a later PR. Using `Time::Zero` as a sentinel value isn't something I love.

I completely agree, `Time::Zero` can be interpreted as "immediate timeout" (i.e. behave as if polling) instead of "infinite timeout". I'd prefer to see `std::optional<Time>` in all our interfaces and use the empty state as "absence of a timeout".

I can work on that on a separate PR after this one.